--- origsrc/dconf-0.22.0/bin/Makefile.am	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/bin/Makefile.am	2015-01-04 13:12:57.985272800 -0600
@@ -11,7 +11,7 @@ dconf_CFLAGS = \
 	-w
 
 dconf_LDADD = \
-	../client/libdconf.so.1		\
+	../client/libdconf.la	\
 	$(gio_LIBS)
 
 dconf_SOURCES = \
--- origsrc/dconf-0.22.0/client/Makefile.am	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/client/Makefile.am	2015-01-04 13:12:57.985272800 -0600
@@ -1,21 +1,8 @@
 include $(top_srcdir)/Makefile.gtester
 
-shlibdir=$(libdir)
-shlib_PROGRAMS = libdconf.so.1.0.0
-nodist_noinst_DATA = libdconf.so.1 libdconf.so
+lib_LTLIBRARIES = libdconf.la
 noinst_LIBRARIES = libdconf-client.a
 
-libdconf.so.1 libdconf.so: libdconf.so.1.0.0
-	$(AM_V_GEN) ln -fs libdconf.so.1.0.0 $@
-
-install-data-hook:
-	ln -fs libdconf.so.1.0.0 $(DESTDIR)$(shlibdir)/libdconf.so.1
-	ln -fs libdconf.so.1.0.0 $(DESTDIR)$(shlibdir)/libdconf.so
-
-uninstall-hook:
-	rm -f $(DESTDIR)$(shlibdir)/libdconf.so.1
-	rm -f $(DESTDIR)$(shlibdir)/libdconf.so
-
 dconfinclude_HEADERS = dconf.h
 dconfclientinclude_HEADERS = dconf-client.h
 dconfclientincludedir = $(dconfincludedir)/client
@@ -24,16 +11,16 @@ libdconf_client_a_CFLAGS = $(gio_CFLAGS)
 libdconf_client_a_SOURCES = \
 	dconf-client.c
 
-libdconf_so_1_0_0_CFLAGS = $(libdconf_client_a_CFLAGS) -fPIC -DPIC
-libdconf_so_1_0_0_LDADD = \
-	../engine/libdconf-engine-shared.a		\
-	../common/libdconf-common-shared.a		\
-	../gdbus/libdconf-gdbus-thread-shared.a		\
-	../gvdb/libgvdb-shared.a			\
-	../shm/libdconf-shm-shared.a			\
+libdconf_la_CFLAGS = $(libdconf_client_a_CFLAGS)
+libdconf_la_LIBADD = \
+	../engine/libdconf-engine-shared.la		\
+	../common/libdconf-common-shared.la		\
+	../gdbus/libdconf-gdbus-thread-shared.la	\
+	../gvdb/libgvdb-shared.la			\
+	../shm/libdconf-shm-shared.la			\
 	$(gio_LIBS)
-libdconf_so_1_0_0_LDFLAGS = -shared -Wl,-soname=libdconf.so.1
-libdconf_so_1_0_0_SOURCES = $(libdconf_client_a_SOURCES)
+libdconf_la_LDFLAGS = -shared -version-number 1:0:0 -no-undefined
+libdconf_la_SOURCES = $(libdconf_client_a_SOURCES)
 
 EXTRA_DIST = dconf.vapi dconf.deps
 
@@ -42,5 +29,3 @@ vapidir = $(datadir)/vala/vapi
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = dconf.pc
-
-CLEANFILES = libdconf.so.1 libdconf.so
--- origsrc/dconf-0.22.0/common/Makefile.am	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/common/Makefile.am	2015-01-04 13:12:57.985272800 -0600
@@ -6,7 +6,8 @@ dconfcommoninclude_HEADERS = \
 	dconf-error.h		\
 	dconf-paths.h
 
-noinst_LIBRARIES = libdconf-common.a libdconf-common-shared.a libdconf-common-hidden.a
+noinst_LIBRARIES = libdconf-common.a
+noinst_LTLIBRARIES = libdconf-common-shared.la libdconf-common-hidden.la
 
 libdconf_common_a_CFLAGS = $(glib_CFLAGS) -DG_LOG_DOMAIN=\"dconf\"
 libdconf_common_a_SOURCES = \
@@ -14,8 +15,8 @@ libdconf_common_a_SOURCES = \
 	dconf-error.c		\
 	dconf-paths.c
 
-libdconf_common_shared_a_CFLAGS = $(libdconf_common_a_CFLAGS) -fPIC -DPIC
-libdconf_common_shared_a_SOURCES = $(libdconf_common_a_SOURCES)
+libdconf_common_shared_la_CFLAGS = $(libdconf_common_a_CFLAGS)
+libdconf_common_shared_la_SOURCES = $(libdconf_common_a_SOURCES)
 
-libdconf_common_hidden_a_CFLAGS = $(libdconf_common_a_CFLAGS) -fPIC -DPIC -fvisibility=hidden
-libdconf_common_hidden_a_SOURCES = $(libdconf_common_a_SOURCES)
+libdconf_common_hidden_la_CFLAGS = $(libdconf_common_a_CFLAGS) -fvisibility=hidden
+libdconf_common_hidden_la_SOURCES = $(libdconf_common_a_SOURCES)
--- origsrc/dconf-0.22.0/configure.ac	2014-09-19 11:23:46.000000000 -0500
+++ src/dconf-0.22.0/configure.ac	2015-01-04 13:12:57.985272800 -0600
@@ -21,7 +21,7 @@ AC_PROG_CC
 AC_CHECK_LIB(c, dlsym, LIBDL="", [AC_CHECK_LIB(dl, dlsym, LIBDL="-ldl")])
 AC_SUBST(LIBDL)
 AM_PROG_VALAC([0.18.0])
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 AC_ARG_ENABLE(man,
               [AS_HELP_STRING([--enable-man],
                               [generate man pages [default=yes]])],,
--- origsrc/dconf-0.22.0/engine/Makefile.am	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/engine/Makefile.am	2015-01-04 13:12:58.000872900 -0600
@@ -1,6 +1,7 @@
 include $(top_srcdir)/Makefile.gtester
 
-noinst_LIBRARIES = libdconf-engine.a libdconf-engine-shared.a
+noinst_LIBRARIES = libdconf-engine.a
+noinst_LTLIBRARIES = libdconf-engine-shared.la
 
 libdconf_engine_a_CFLAGS = $(glib_CFLAGS) -DG_LOG_DOMAIN=\"dconf\"
 libdconf_engine_a_SOURCES = \
@@ -16,5 +17,5 @@ libdconf_engine_a_SOURCES = \
 	dconf-engine.h			\
 	dconf-engine.c
 
-libdconf_engine_shared_a_CFLAGS = $(libdconf_engine_a_CFLAGS) -fPIC -DPIC
-libdconf_engine_shared_a_SOURCES = $(libdconf_engine_a_SOURCES)
+libdconf_engine_shared_la_CFLAGS = $(libdconf_engine_a_CFLAGS)
+libdconf_engine_shared_la_SOURCES = $(libdconf_engine_a_SOURCES)
--- origsrc/dconf-0.22.0/gdbus/Makefile.am	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/gdbus/Makefile.am	2015-01-04 13:12:58.000872900 -0600
@@ -2,20 +2,21 @@ include $(top_srcdir)/Makefile.gtester
 
 noinst_LIBRARIES = \
 	libdconf-gdbus-thread.a			\
-	libdconf-gdbus-thread-shared.a		\
-	libdconf-gdbus-filter.a			\
-	libdconf-gdbus-filter-shared.a
+	libdconf-gdbus-filter.a
+noinst_LTLIBRARIES = \
+	libdconf-gdbus-thread-shared.la		\
+	libdconf-gdbus-filter-shared.la
 
 libdconf_gdbus_thread_a_CFLAGS = $(gio_CFLAGS) -DG_LOG_DOMAIN=\"dconf\"
 libdconf_gdbus_thread_a_SOURCES = \
 	dconf-gdbus-thread.c
 
-libdconf_gdbus_thread_shared_a_CFLAGS = $(libdconf_gdbus_thread_a_CFLAGS) -fPIC -DPIC
-libdconf_gdbus_thread_shared_a_SOURCES = $(libdconf_gdbus_thread_a_SOURCES)
+libdconf_gdbus_thread_shared_la_CFLAGS = $(libdconf_gdbus_thread_a_CFLAGS)
+libdconf_gdbus_thread_shared_la_SOURCES = $(libdconf_gdbus_thread_a_SOURCES)
 
 libdconf_gdbus_filter_a_CFLAGS = $(gio_CFLAGS) -DG_LOG_DOMAIN=\"dconf\"
 libdconf_gdbus_filter_a_SOURCES = \
 	dconf-gdbus-filter.c
 
-libdconf_gdbus_filter_shared_a_CFLAGS = $(libdconf_gdbus_filter_a_CFLAGS) -fPIC -DPIC
-libdconf_gdbus_filter_shared_a_SOURCES = $(libdconf_gdbus_filter_a_SOURCES)
+libdconf_gdbus_filter_shared_la_CFLAGS = $(libdconf_gdbus_filter_a_CFLAGS)
+libdconf_gdbus_filter_shared_la_SOURCES = $(libdconf_gdbus_filter_a_SOURCES)
--- origsrc/dconf-0.22.0/gsettings/Makefile.am	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/gsettings/Makefile.am	2015-01-04 13:12:58.016472900 -0600
@@ -1,18 +1,18 @@
 include $(top_srcdir)/Makefile.gtester
 
-giomodules_PROGRAMS = libdconfsettings.so
+giomodules_LTLIBRARIES = libdconfsettings.la
 
-libdconfsettings_so_LDADD = \
-	../engine/libdconf-engine-shared.a		\
-	../gvdb/libgvdb-shared.a			\
-	../gdbus/libdconf-gdbus-thread-shared.a		\
-	../common/libdconf-common-hidden.a		\
-	../shm/libdconf-shm-shared.a			\
+libdconfsettings_la_LIBADD = \
+	../engine/libdconf-engine-shared.la		\
+	../gvdb/libgvdb-shared.la			\
+	../gdbus/libdconf-gdbus-thread-shared.la	\
+	../common/libdconf-common-hidden.la		\
+	../shm/libdconf-shm-shared.la			\
 	$(gio_LIBS)
 
-libdconfsettings_so_CFLAGS = $(gio_CFLAGS) -fPIC -DPIC -DG_LOG_DOMAIN=\"dconf\"
-libdconfsettings_so_LDFLAGS = -shared
-libdconfsettings_so_SOURCES = \
+libdconfsettings_la_CFLAGS = $(gio_CFLAGS) -DG_LOG_DOMAIN=\"dconf\"
+libdconfsettings_la_LDFLAGS = -shared -module -avoid-version -no-undefined
+libdconfsettings_la_SOURCES = \
 	dconfsettingsbackend.c
 
 uninstall-hook:
--- origsrc/dconf-0.22.0/gvdb/Makefile.am	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/gvdb/Makefile.am	2015-01-04 13:12:58.016472900 -0600
@@ -1,6 +1,7 @@
 include $(top_srcdir)/Makefile.gtester
 
-noinst_LIBRARIES = libgvdb.a libgvdb-shared.a
+noinst_LIBRARIES = libgvdb.a
+noinst_LTLIBRARIES = libgvdb-shared.la
 
 libgvdb_a_CFLAGS = $(glib_CFLAGS) -DG_LOG_DOMAIN=\"gvdb\ \(via\ dconf\)\"
 libgvdb_a_SOURCES = \
@@ -10,5 +11,5 @@ libgvdb_a_SOURCES = \
 	gvdb-reader.h	\
 	gvdb-reader.c
 
-libgvdb_shared_a_CFLAGS = $(libgvdb_a_CFLAGS) -fPIC -DPIC
-libgvdb_shared_a_SOURCES = $(libgvdb_a_SOURCES)
+libgvdb_shared_la_CFLAGS = $(libgvdb_a_CFLAGS)
+libgvdb_shared_la_SOURCES = $(libgvdb_a_SOURCES)
--- origsrc/dconf-0.22.0/shm/Makefile.am	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/shm/Makefile.am	2015-01-04 13:12:58.016472900 -0600
@@ -1,11 +1,12 @@
 include $(top_srcdir)/Makefile.gtester
 
-noinst_LIBRARIES = libdconf-shm.a libdconf-shm-shared.a
+noinst_LIBRARIES = libdconf-shm.a
+noinst_LTLIBRARIES = libdconf-shm-shared.la
 
 libdconf_shm_a_CFLAGS = $(glib_CFLAGS) -DG_LOG_DOMAIN=\"dconf\"
 libdconf_shm_a_SOURCES = \
 	dconf-shm.h		\
 	dconf-shm.c
 
-libdconf_shm_shared_a_CFLAGS = $(libdconf_shm_a_CFLAGS) -fPIC -DPIC
-libdconf_shm_shared_a_SOURCES = $(libdconf_shm_a_SOURCES)
+libdconf_shm_shared_la_CFLAGS = $(libdconf_shm_a_CFLAGS)
+libdconf_shm_shared_la_SOURCES = $(libdconf_shm_a_SOURCES)
