--- origsrc/dconf-0.22.0/tests/engine.c	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/tests/engine.c	2015-01-04 13:16:04.982798900 -0600
@@ -13,6 +13,9 @@
 #include <stdio.h>
 #include <dlfcn.h>
 #include <math.h>
+#ifdef __CYGWIN__
+#include <sys/cygwin.h>
+#endif
 
 /* Interpose to catch fopen("/etc/dconf/profile/user") */
 static const gchar *filename_to_replace;
@@ -23,6 +26,12 @@ fopen (const char *filename,
        const char *mode)
 {
   static FILE * (*real_fopen) (const char *, const char *);
+#ifdef __CYGWIN__
+  static void *RTLD_NEXT = NULL;
+
+  if (!RTLD_NEXT)
+    RTLD_NEXT = dlopen ("/usr/bin/cygwin1.dll", 0);
+#endif
 
   if (!real_fopen)
     real_fopen = dlsym (RTLD_NEXT, "fopen");
@@ -1740,6 +1749,10 @@ test_sync (void)
 int
 main (int argc, char **argv)
 {
+#ifdef __CYGWIN__
+  cygwin_internal (CW_HOOK, "fopen", fopen);
+#endif
+
   g_setenv ("XDG_RUNTIME_DIR", "/RUNTIME/", TRUE);
   g_setenv ("XDG_CONFIG_HOME", "/HOME/.config", TRUE);
   g_unsetenv ("DCONF_PROFILE");
--- origsrc/dconf-0.22.0/tests/shm.c	2014-03-03 14:58:21.000000000 -0600
+++ src/dconf-0.22.0/tests/shm.c	2015-01-04 13:16:04.998398900 -0600
@@ -10,6 +10,9 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <dlfcn.h>
+#ifdef __CYGWIN__
+#include <sys/cygwin.h>
+#endif
 
 #include "../shm/dconf-shm.h"
 #include "tmpdir.h"
@@ -91,6 +94,12 @@ ssize_t
 pwrite (int fd, const void *buf, size_t count, off_t offset)
 {
   static ssize_t (* real_pwrite) (int, const void *, size_t, off_t);
+#ifdef __CYGWIN__
+  static void *RTLD_NEXT = NULL;
+
+  if (!RTLD_NEXT)
+    RTLD_NEXT = dlopen ("/usr/bin/cygwin1.dll", 0);
+#endif
 
   if (!real_pwrite)
     real_pwrite = dlsym (RTLD_NEXT, "pwrite");
@@ -143,6 +152,10 @@ main (int argc, char **argv)
   gchar *temp;
   gint status;
 
+#ifdef __CYGWIN__
+  cygwin_internal (CW_HOOK, "pwrite", pwrite);
+#endif
+
   temp = dconf_test_create_tmpdir ();
 
   g_setenv ("XDG_RUNTIME_DIR", temp, TRUE);
