--- origsrc/dconf-0.6/bin/Makefile.am	2010-08-03 15:24:14.000000000 -0500
+++ src/dconf-0.6/bin/Makefile.am	2010-12-29 20:50:14.038081700 -0600
@@ -4,5 +4,5 @@ INCLUDES = -I$(top_srcdir)/common -I$(to
 bin_PROGRAMS = dconf
 
 dconf_VALAFLAGS = --vapidir ../client --pkg=gio-2.0 --pkg=posix --pkg=dconf
-dconf_LDADD = $(gio_LIBS) ../client/libdconf.so.0
+dconf_LDADD = $(gio_LIBS) ../client/libdconf.la
 dconf_SOURCES = dconf.vala dconf-update.vala ../gvdb/gvdb-builder.c gvdb.vapi fixes.vapi
--- origsrc/dconf-0.6/client/Makefile.am	2010-12-21 11:25:23.000000000 -0600
+++ src/dconf-0.6/client/Makefile.am	2010-12-29 20:50:14.038081700 -0600
@@ -3,35 +3,24 @@ CFLAGS += -Wno-error
 INCLUDES = -I$(top_srcdir)/common -I$(top_srcdir)/gvdb -I$(top_srcdir)/engine $(gio_CFLAGS)
 
 shlibdir=$(libdir)
-shlib_PROGRAMS = libdconf.so.0.0.0
-nodist_noinst_DATA = libdconf.so.0 libdconf.so
-
-libdconf.so.0 libdconf.so: libdconf.so.0.0.0
-	$(AM_V_GEN) ln -fs libdconf.so.0.0.0 $@
-
-install-data-hook:
-	ln -fs libdconf.so.0.0.0 $(DESTDIR)$(shlibdir)/libdconf.so.0
-	ln -fs libdconf.so.0.0.0 $(DESTDIR)$(shlibdir)/libdconf.so
-
-uninstall-hook:
-	rm -f $(DESTDIR)$(shlibdir)/libdconf.so.0
-	rm -f $(DESTDIR)$(shlibdir)/libdconf.so
+lib_LTLIBRARIES = libdconf.la
 
 dconfinclude_HEADERS = \
 	dconf-client.h	\
 	dconf.h
 
-libdconf_so_0_0_0_LDFLAGS = -shared -Wl,-soname=libdconf.so.0 $(gio_LIBS)
-libdconf_so_0_0_0_SOURCES = \
+libdconf_la_LDFLAGS = -version-info 0:0:0 -no-undefined
+libdconf_la_LIBADD = $(gio_LIBS)
+libdconf_la_SOURCES = \
 	../common/dconf-shmdir.c	\
 	../common/dconf-paths.c		\
 	../engine/dconf-engine.c	\
 	../gvdb/gvdb-reader.c		\
 	dconf-client.vala engine.vapi
-libdconf_so_0_0_0_VALAFLAGS = --library dconf --pkg=gio-2.0
+libdconf_la_VALAFLAGS = --library dconf --pkg=gio-2.0
 
 EXTRA_DIST = dconf.vapi extra-docs.c
-dconf.vapi: libdconf.so.0
+dconf.vapi: libdconf.la
 
 dconf.deps:
 	$(AM_V_GEN) echo gio-2.0 > dconf.deps
--- /dev/null
+++ src/dconf-0.6/client/libdconf_la_vala.stamp
@@ -0,0 +1,1
+timestamp
--- origsrc/dconf-0.6/configure.ac	2010-12-21 11:33:48.000000000 -0600
+++ src/dconf-0.6/configure.ac	2010-12-29 20:50:14.053681800 -0600
@@ -12,6 +12,7 @@ AM_SILENT_RULES([yes])
 
 # Check for programs
 AC_PROG_CC
+AC_PROG_LIBTOOL
 AM_PROG_VALAC([0.9.5])
 
 # Gtk-doc support
--- origsrc/dconf-0.6/editor/Makefile.am	2010-12-18 10:36:32.000000000 -0600
+++ src/dconf-0.6/editor/Makefile.am	2010-12-29 20:51:29.261412900 -0600
@@ -3,5 +3,5 @@ bin_PROGRAMS = dconf-editor
 AM_CFLAGS = $(gtk_CFLAGS) $(libxml_CFLAGS) -I$(top_srcdir)/common -I$(top_srcdir)/client 
 AM_VALAFLAGS = --vapidir ../client --pkg gtk+-3.0 --pkg libxml-2.0 --pkg dconf
 CFLAGS += -Wno-error
-dconf_editor_LDADD = ../client/libdconf.so.0 $(gtk_LIBS) $(gee_LIBS) $(libxml_LIBS)
+dconf_editor_LDADD = ../client/libdconf.la $(gtk_LIBS) $(gee_LIBS) $(libxml_LIBS)
 dconf_editor_SOURCES = dconf-editor.vala dconf-model.vala dconf-schema.vala dconf-view.vala
--- origsrc/dconf-0.6/gsettings/Makefile.am	2010-12-21 11:26:46.000000000 -0600
+++ src/dconf-0.6/gsettings/Makefile.am	2010-12-29 20:50:14.069281800 -0600
@@ -1,11 +1,11 @@
 AM_CFLAGS = -std=c89 -Wall -Wmissing-prototypes -Wwrite-strings -fPIC -DPIC
 INCLUDES = -I$(top_srcdir)/common -I$(top_srcdir)/gvdb -I$(top_srcdir)/engine $(gio_CFLAGS)
 
-giomodules_PROGRAMS = libdconfsettings.so
+giomodules_LTLIBRARIES = libdconfsettings.la
 
-libdconfsettings_so_LDADD = $(gio_LIBS)
-libdconfsettings_so_LDFLAGS = -module -avoid-version -shared
-libdconfsettings_so_SOURCES = \
+libdconfsettings_la_LIBADD = $(gio_LIBS)
+libdconfsettings_la_LDFLAGS = -module -avoid-version -no-undefined
+libdconfsettings_la_SOURCES = \
 	../engine/dconf-engine.c	\
 	../common/dconf-shmdir.c	\
 	../gvdb/gvdb-reader.c		\
